### ISSUE TITLE
[Jetpack Migration] Migration main flow is broken on beta version 21.8

### ISSUE THREAD

**Issue by tiagomar** [wordpress-mobile/WordPress-iOS#20207]
Migration main flow is broken on `21.8` when Jetpack app is not previously installed. The same flow works correctly on `21.7`.

| ✅ **21.7** | ❌ **21.8** |
| - | - |
| <video width="300" src="https://user-images.githubusercontent.com/42008628/221072872-8484d0e0-6f94-438c-8b0e-740403cb8e60.mov"> | <video width="300" src="https://user-images.githubusercontent.com/42008628/221072753-99ac0622-4d45-43dc-8239-d7712cd98b65.mov"> |

### Expected behavior
When migrating from WordPress to Jetpack, the user should land on the Jetpack app already logged in.

### Actual behavior
When migrating from WordPress to Jetpack, the user lands on the Jetpack app landing page, logged out.

### Steps to reproduce the behavior
Precondition: Jetpack app is not installed
1. Install WordPress app, 21.8 - currently available on TestFlight.
2. Log into WordPress.
3. Tap the `Try the new Jetpack app` button from a bottom sheet or overlay.
4. After being redirected to the App Store, install and open the Jetpack app.
5. ❌ Jetpack landing screen is displayed and the user is not logged in.

**Comment by dvdchr** [comment]
Hey @tiagomar 👋🏼 , thanks for the report. I've taken a look at this, and it looks like this was due to a data model versioning mismatch:

```
Error restoring database: Error Domain=NSCocoaErrorDomain Code=134100 "The managed object model version used to open the persistent store is incompatible with the one that was used to create the persistent store."
```

In 21.8, a new field was added through #20053 which bumped the data model version from 145 to 146. The data migration process currently does not check if the copied data model needs to be migrated first. 

The good news is that this is _partially_ solvable: We should be able to detect if the copied database needs to be migrated, and migrate them through `NSMigrationManager`. But I think this only works when we're advancing the model version (from a lower to a higher one). I'm not sure if it's possible to "downgrade" the model version 🤔.

Also, probably worth noting that I've found a reference from [a 2020 article](https://bipsync.com/blog/dealing-with-manual-core-data-migrations/) that explores the manual migration topic. But I haven't really looked into this yet.

#### TL;DR

It's possible to fix this, but it'll only apply when Jetpack has a higher schema version than WordPress. I'm not sure it can work the other way around. Here's a comparison before and after the fix:

WP | JP | Before the fix | After the fix
-|-|-|-
21.7 | 21.7 | ✅ Can migrate | ✅ Can migrate
21.8 | 21.8 | ✅ Can migrate | ✅ Can migrate
21.7 | 21.8 | ❌ Error; show sign-in | ✅ Can migrate
21.8 | 21.7 | ❌ Error; show sign-in | ❌ Error; show sign-in

I'm looping in @wargcm for more context and will discuss the next steps with him.

**Comment by salimbraksa** [comment]
>In 21.8, a new field was added through https://github.com/wordpress-mobile/WordPress-iOS/pull/20053 which bumped the data model version from 145 to 146.

@dvdchr I want to clarify here that the PR added a brand new entity **BlockedSite**.

**Comment by dvdchr** [comment]
Update: This will be addressed via #20217 (courtesy of @wargcm). The PR targets `21.7.2` (so that we can handle the downgrade case), and will also be included in `21.8`. Here's a table outlining various setups, before and after the fix:

WP | JP | Before the fix | After the fix
-|-|-|-
21.7 | 21.7 | ✅ Can migrate | ✅ Can migrate
21.8 | 21.8 | ✅ Can migrate | ✅ Can migrate
21.7 | 21.8 | ❌ Error; show sign-in | ✅ Can migrate
21.8 | 21.7.2 | ❌ Error; show sign-in | ✅ Can migrate
21.8 | ≤21.7.1 | ❌ Error; show sign-in | ❌ Error; show sign-in

**Comment by dvdchr** [comment]
Closing as this should be resolved by #20217, which will be merged to 21.7.2 and 21.8 beta. 

Feel free to reopen if you still encounter this issue! 🙂
